### PR TITLE
AR-240. added adoptive parent info

### PR DESCRIPTION
**Pull request by savelich5**
Information about adoptive parent was added to the animal model.
Validation was added as well.

### CODE REVIEW

File: Backend/AnimalRescue/AnimalRescue.API/Models/Animals/AnimalCreateUpdateModel.cs [hunk 1]
```diff
@@ -77,6 +74,51 @@ public class AnimalCreateUpdateModel
...
+        public string AdoptiveName { get; set; }
+
+        [RegularExpression(@"^[0-9\-\+]{9,15}$", ErrorMessage = "Incorrect phone format. Phone number must contains only digits.")]
+        [JsonPropertyName(animal.AdoptivePhone)]
+        [JsonProperty(animal.AdoptivePhone)]
+        public string AdoptivePhone { get; set; }
+
+        [JsonPropertyName(animal.AdoptionContractFile)]
+        [JsonProperty(animal.AdoptionContractFile)]
+        public IFormFile AdoptionContractFile { get; set; }
```
Comment (DragonTM) [inline Backend/AnimalRescue/AnimalRescue.API/Models/Animals/AnimalCreateUpdateModel.cs:90]: Please review that the property is handled in the controller

Comment (savelich5) [reply Backend/AnimalRescue/AnimalRescue.API/Models/Animals/AnimalCreateUpdateModel.cs:90]: fixed;

File: Backend/AnimalRescue/AnimalRescue.API/Controllers/AnimalsController.cs [hunk 2]
```diff
@@ -125,6 +129,7 @@ public async Task UpdateAsync([BindRequired, FromRoute] Guid id, [FromForm] Anim
 
             var animalModel = _mapper.Map<AnimalCreateUpdateModel, AnimalModel>(animalUpdateModel);
             animalModel.Id = id;
+            animalModel.AdoptionContractFileId = await HandleAdoptionContractFileAsync(animalUpdateModel.AdoptionContractFile);
```
Comment (DragonTM) [inline Backend/AnimalRescue/AnimalRescue.API/Controllers/AnimalsController.cs]: Does it mean that the user has to upload a new file every time they update an animal card?

Comment (savelich5) [reply Backend/AnimalRescue/AnimalRescue.API/Controllers/AnimalsController.cs]: It was the bug. Fixed.

**Review by DragonTM — APPROVED** [review]

**Pull request merged**